ipams/remote/remote.go

```diff
@@ -93,6 +93,8 @@ func (a *allocator) RequestAddress(poolID string, address net.IP, options map[st
 	}
 	if res.Address != "" {
 		retAddress, err = types.ParseCIDR(res.Address)
+	} else {
+		return nil, nil, ipamapi.ErrNoAvailableIPs
```

Comment:
The allocation errors (no available ips, ip already allocated, ...) should be returned by the remote driver in `RequestAddress()` response `RequestAddressResponse.Response.Error`

If the generic call fails `a.call` will return the request's error (all responses embed `api.Response`).